benchmarks/smallmatvec/manual_versions/libxsmm/kdriver.f90

```diff
@@ -0,0 +1,186 @@
+      allocate( theta_adv_term_data(undf_any_space_1_theta_adv_term) )
+      allocate( x_data(undf_any_space_2_x) )
+      allocate( ptheta_2_local_stencil( ndf_any_space_1_theta_adv_term, ndf_any_space_2_x, ncell_3d) )
+      allocate( nlayers_first(nlayers, ndf_any_space_2_x, ndf_any_space_1_theta_adv_term, ncell) )
+      allocate( ans_data(undf_any_space_1_theta_adv_term) )
+      allocate( ncp_colour(ncolour) )
+
+      ! Populate arrays
+      do i2 = 0, nsize - 1
+        do i1 = 0, nsize - 1
+          ! Populate  map_any_space_1_theta_adv_term (8 vertices)
+          do el = 1, 2
+            map_any_space_1_theta_adv_term( 1 + 4*(el-1), (i2 * nsize) + i1 + 1 ) = (i2 * (nsize+1) + i1) * (nlayers + 1) + el
+            map_any_space_1_theta_adv_term( 2 + 4*(el-1), (i2 * nsize) + i1 + 1 ) = (i2 * (nsize+1) + i1+1) * (nlayers + 1) + el
+            map_any_space_1_theta_adv_term( 3 + 4*(el-1), (i2 * nsize) + i1 + 1 ) = ((i2+1) * (nsize+1) + i1+1) * (nlayers + 1) + el
+            map_any_space_1_theta_adv_term( 4 + 4*(el-1), (i2 * nsize) + i1 + 1 ) = ((i2+1) * (nsize+1) + i1) * (nlayers + 1) + el
+          enddo
+
+          !Populate  map_any_space_2_x (6 surfaces)
+          map_any_space_2_x(1, (i2 * nsize) + i1 + 1 ) = i2 * ((3 * nsize * nlayers) + nsize + nlayers) + i1 * (3*nlayers + 1) + 1
+          map_any_space_2_x(2, (i2 * nsize) + i1 + 1 ) = i2 * ((3 * nsize * nlayers) + nsize + nlayers) + i1 * (3*nlayers + 1) + nlayers + 1
+          if (i1.eq.nsize-1) then
+            map_any_space_2_x(3, (i2 * nsize) + i1 + 1 ) = i2 * ((3 * nsize * nlayers) + nsize + nlayers) + (i1+1) * (3*nlayers + 1) + 1
+          else
+            map_any_space_2_x(3, (i2 * nsize) + i1 + 1 ) = i2 * ((3 * nsize * nlayers) + nsize + nlayers) + (i1+1) * (3*nlayers + 1) + nlayers + 1
+          endif
+          if (i2.eq.nsize-1) then
+            map_any_space_2_x(4, (i2 * nsize) + i1 + 1 ) = (i2+1) * ((3 * nsize * nlayers) + nsize + nlayers) + i1 * nlayers + 1
+          else
+            map_any_space_2_x(4, (i2 * nsize) + i1 + 1 ) = (i2+1) * ((3 * nsize * nlayers) + nsize + nlayers) + i1 * (3*nlayers + 1) + 1
+          endif
+          map_any_space_2_x(5, (i2 * nsize) + i1 + 1 ) = i2 * ((3 * nsize * nlayers) + nsize + nlayers) + i1 * (3*nlayers + 1) + 2*nlayers + 1
+          map_any_space_2_x(6, (i2 * nsize) + i1 + 1 ) = i2 * ((3 * nsize * nlayers) + nsize + nlayers) + i1 * (3*nlayers + 1) + 2*nlayers + 2
+        enddo
+      enddo
+
+      
+      if (traverse.eq."colouring") then
+        ! Number of cells per color = num times per row * num times per column
+        ncp_colour(1) = ((nsize/2)+mod(nsize,2)) * ((nsize/2)+mod(nsize,2))
+        ncp_colour(2) = ( nsize/2 )              * ((nsize/2)+mod(nsize,2))
+        ncp_colour(3) = ((nsize/2)+mod(nsize,2)) * (nsize/2)
+        ncp_colour(4) = ( nsize/2 )              * (nsize/2)
+
+        allocate(cmap(ncolour,maxval(ncp_colour))) !color map (colour, cell number)
```

Comment:
rm commented code?

Reply:
I've been making some general changes and can't see commented code so I guess I've done this :-)